commons/src/main/java/org/infinispan/commons/util/CloseableIteratorAdapter.java

```diff
@@ -0,0 +1,30 @@
+package org.infinispan.commons.util;
+
+import java.util.Iterator;
+
+/**
+ * Adapts regular {@link Iterator} to {@link CloseableIterator}
+ *
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public final class CloseableIteratorAdapter<E> implements CloseableIterator<E> {
```

Comment:
This class is already implemented https://github.com/infinispan/infinispan/blob/master/commons/src/main/java/org/infinispan/commons/util/IteratorAsCloseableIterator.java

You can tweak that one to allow \<? extends E\> though.  Although I am actually moving this in https://github.com/infinispan/infinispan/pull/3569/files#diff-98ed1727054c0e59fa45ca9079aa4d6eR39